Report: program compilation error when using varchar as primary key

**Describe the bug**

Creating a table using `VARCHAR` or `VARCHAR(n)` results in rust error

```

...Beginning of the error...go_workspace/project0190a407-1ebe-7421-98f5-66d88c2eefc1)
error[E0507]: cannot move out of `t_2` which is behind a shared reference
   --> project0190a407-1ebe-7421-98f5-66d88c2eefc1/src/main.rs:174:23
    |
174 |             Tup1::new((*t_2).0)
    |                       ^^^^^^^^ move occurs because `t_2.0` has type `std::option::Option<std::string::String>`, which does not implement the `Copy` trait

error[E0507]: cannot move out of `t_3` which is behind a shared reference
   --> project0190a407-1ebe-7421-98f5-66d88c2eefc1/src/main.rs:177:23
    |
177 |             Tup1::new((*t_3).0)
    |                       ^^^^^^^^ move occurs because `t_3.0` has type `std::option::Option<std::string::String>`, which does not implement the `Copy` trait

For more information about this error, try `rustc --explain E0507`.
error: could not compile `project0190a407-1ebe-7421-98f5-66d88c2eefc1` (bin "project0190a407-1ebe-7421-98f5-66d88c2eefc1") due to 2 previous errors

exit code: exit status: 101
```

**To Reproduce**
<!-- Steps to reproduce the behavior e.g., Curl commands or steps in the UI such as:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error -->

1. Create SQL program that includes a table with `VARCHAR` as primary key
```
CREATE TABLE varchar_pk (
    pk VARCHAR PRIMARY KEY
)
```
2. SQL program successfully saves
3. Short moment later, Web UI reports error

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
To be able to use `VARCHAR` as primary key


**Context:**

 - Feldera Version: 0.20.0
 - Environment: Docker
 - Browser: Chrome

--- a/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/circuit/operator/DBSPSourceMapOperator.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/circuit/operator/DBSPSourceMapOperator.java
@@ -114,7 +114,7 @@ public final class DBSPSourceMapOperator extends DBSPSourceTableOperator {
         DBSPExpression[] fields = new DBSPExpression[this.keyFields.size()];
         int insertAt = 0;
         for (int index: this.keyFields) {
-            fields[insertAt++] = var.deref().field(index);
+            fields[insertAt++] = var.deref().field(index).applyCloneIfNeeded();
         }
         DBSPExpression tuple = new DBSPTupleExpression(fields);
         return tuple.closure(var.asParameter());
@@ -126,7 +126,7 @@ public final class DBSPSourceMapOperator extends DBSPSourceTableOperator {
         DBSPExpression[] fields = new DBSPExpression[this.keyFields.size()];
         int insertAt = 0;
         for (int index: this.keyFields) {
-            fields[insertAt++] = var.deref().field(index);
+            fields[insertAt++] = var.deref().field(index).applyCloneIfNeeded();
         }
         DBSPExpression tuple = new DBSPTupleExpression(fields);
         return tuple.closure(var.asParameter());

--- a/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/CatalogTests.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/CatalogTests.java
@@ -18,6 +18,17 @@ public class CatalogTests extends BaseSQLTests {
     }
 
     @Test
+    public void issue2028() {
+        String sql = """
+                CREATE TABLE varchar_pk (
+                    pk VARCHAR PRIMARY KEY
+                );
+                
+                CREATE VIEW V AS SELECT * FROM varchar_pk;""";
+        this.compileRustTestCase(sql);
+    }
+
+    @Test
     public void issue1894() {
         String sql = """
                 CREATE TYPE trans AS (

